From 63535ab36f601964019cc904db069e752670c8c3 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Sun, 5 Mar 2023 10:14:04 +0200
Subject: [PATCH 215/229] arm64: dts: imx8dxl-v2x: split carrier into separate
 dtsi

Split out the carrier into dtsi to simplify introducing new revisions of
SoM / carrier independently.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 .../dts/freescale/imx8dxl-sr-v2x-gateway.dtsi | 559 ++++++++++++++++++
 arch/arm64/boot/dts/freescale/imx8dxl-v2x.dts | 557 +----------------
 2 files changed, 563 insertions(+), 553 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8dxl-sr-v2x-gateway.dtsi

diff --git a/arch/arm64/boot/dts/freescale/imx8dxl-sr-v2x-gateway.dtsi b/arch/arm64/boot/dts/freescale/imx8dxl-sr-v2x-gateway.dtsi
new file mode 100644
index 000000000000..6c7c013e86a2
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8dxl-sr-v2x-gateway.dtsi
@@ -0,0 +1,559 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright 2019~2020 NXP
+ * Copyright 2022-2023 Josua Mayer <josua@solid-run.com>
+ */
+
+/ {
+	/*lte_vbat_reg: regulator-lte-vbat {
+		compatible = "regulator-fixed";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_lte_vbat>;
+		regulator-name = "lte-vbat";
+		regulator-min-microvolt = <3600000>;
+		regulator-max-microvolt = <3600000>;
+		gpio = <&lsio_gpio0 14 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		regulator-always-on;
+	};*/
+
+	regulator_1v1: regulator-1v1 {
+		compatible = "regulator-fixed";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_reg_1v1>;
+		regulator-name = "1v1";
+		regulator-min-microvolt = <1100000>;
+		regulator-max-microvolt = <1100000>;
+		gpio = <&lsio_gpio4 5 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		regulator-always-on;
+	};
+
+	regulator_flexcan1_standby: regulator-flexcan1-standby {
+		compatible = "regulator-fixed";
+		regulator-name = "flexcan1-standby";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&tca6408 2 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
+	regulator_flexcan2_standby: regulator-flexcan2-standby {
+		compatible = "regulator-fixed";
+		regulator-name = "flexcan2-standby";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&tca6408 3 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
+	usdhc3_pwrseq: usdhc3-pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		reset-gpios = <&lsio_gpio0 15 GPIO_ACTIVE_LOW>;
+	};
+};
+
+&eqos {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_eqos>;
+	phy-connection-type = "rgmii-id";
+	status = "okay";
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&flexcan1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flexcan1>;
+	xceiver-supply = <&regulator_flexcan1_standby>;
+	status = "okay";
+
+	can-transceiver {
+		/* TJA1048 */
+		max-bitrate = <5000000>;
+	};
+};
+
+&flexcan2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flexcan2>;
+	xceiver-supply = <&regulator_flexcan2_standby>;
+	status = "okay";
+
+	can-transceiver {
+		/* TJA1048 */
+		max-bitrate = <5000000>;
+	};
+};
+
+&iomuxc {
+		pinctrl_eqos: eqosgrp {
+			fsl,pins = <
+				/* MDIO to Switch */
+				/* enet0 mdio pads supplied with 3.3v */
+				/* IMX8DXL_COMP_CTL_GPIO_1V8_3V3_GPIOCT */
+				IMX8DXL_ENET0_MDC_CONN_EQOS_MDC				0x06000020
+				IMX8DXL_ENET0_MDIO_CONN_EQOS_MDIO			0x06000020
+				/* RGMII to Switch */
+				/* ENET1 pad supplied with 3.3V */
+				/*IMX8DXL_COMP_CTL_GPIO_1V8_3V3_GPIORHB		0x000514a0*/
+				IMX8DXL_ENET1_RGMII_TX_CTL_CONN_EQOS_RGMII_TX_CTL	0x06000020
+				IMX8DXL_ENET1_RGMII_TXC_CONN_EQOS_RGMII_TXC		0x06000020
+				IMX8DXL_ENET1_RGMII_TXD0_CONN_EQOS_RGMII_TXD0		0x06000020
+				IMX8DXL_ENET1_RGMII_TXD1_CONN_EQOS_RGMII_TXD1		0x06000020
+				IMX8DXL_ENET1_RGMII_TXD2_CONN_EQOS_RGMII_TXD2		0x06000020
+				IMX8DXL_ENET1_RGMII_TXD3_CONN_EQOS_RGMII_TXD3		0x06000020
+				IMX8DXL_ENET1_RGMII_RXC_CONN_EQOS_RGMII_RXC		0x06000020
+				IMX8DXL_ENET1_RGMII_RX_CTL_CONN_EQOS_RGMII_RX_CTL	0x06000020
+				IMX8DXL_ENET1_RGMII_RXD0_CONN_EQOS_RGMII_RXD0		0x06000020
+				IMX8DXL_ENET1_RGMII_RXD1_CONN_EQOS_RGMII_RXD1		0x06000020
+				IMX8DXL_ENET1_RGMII_RXD2_CONN_EQOS_RGMII_RXD2		0x06000020
+				IMX8DXL_ENET1_RGMII_RXD3_CONN_EQOS_RGMII_RXD3		0x06000020
+			>;
+		};
+
+		pinctrl_flexcan1: flexcan1grp {
+			fsl,pins = <
+				IMX8DXL_FLEXCAN0_TX_ADMA_FLEXCAN0_TX	0x00000021
+				IMX8DXL_FLEXCAN0_RX_ADMA_FLEXCAN0_RX	0x00000021
+			>;
+		};
+
+		pinctrl_flexcan2: flexcan2grp {
+			fsl,pins = <
+				IMX8DXL_FLEXCAN1_TX_ADMA_FLEXCAN1_TX	0x00000021
+				IMX8DXL_FLEXCAN1_RX_ADMA_FLEXCAN1_RX	0x00000021
+			>;
+		};
+
+		pinctrl_i2c2_gpio_rst: i2c2gpiorstgrp {
+			fsl,pins = <
+				/* gpio-expander reset: io without pull-up */
+				IMX8DXL_SCU_GPIO0_00_LSIO_GPIO2_IO03		0x0000061
+			>;
+		};
+
+		pinctrl_i2c2_gpio_int: i2c2gpiointgrp {
+			fsl,pins = <
+				/* gpio-expander interrupt: io without pull-up */
+				/* not usable, because this pin does not support lsio gpio mode :( */
+				IMX8DXL_SCU_GPIO0_01_SCU_GPIO0_IO01		0x0000061
+			>;
+		};
+
+		pinctrl_i2c3: i2c3grp {
+			fsl,pins = <
+				IMX8DXL_SPI1_CS0_ADMA_I2C3_SDA		0x06000021
+				IMX8DXL_SPI1_SDI_ADMA_I2C3_SCL		0x06000021
+			>;
+		};
+
+		pinctrl_lpuart1: lpuart1grp {
+			fsl,pins = <
+				IMX8DXL_UART1_RX_ADMA_UART1_RX			0x06000020
+				IMX8DXL_UART1_TX_ADMA_UART1_TX			0x06000020
+				IMX8DXL_UART1_CTS_B_ADMA_UART1_CTS_B	0x06000020
+				IMX8DXL_UART1_RTS_B_ADMA_UART1_RTS_B	0x06000020
+			>;
+		};
+
+		pinctrl_lte: ltegrp {
+			fsl,pins = <
+				/* modem RESET_N: io open drain drive 2mA */
+				IMX8DXL_ADC_IN3_LSIO_GPIO1_IO11	0x2000061
+
+				/* modem PWRKEY: io open drain drive 2mA */
+				IMX8DXL_ADC_IN2_LSIO_GPIO1_IO12	0x2000061
+			>;
+		};
+
+		pinctrl_lte_vbat: ltevbatgrp {
+			fsl,pins = <
+				/* RF_PWR: io without pull-up */
+				IMX8DXL_SPI3_SDO_LSIO_GPIO0_IO14	0x0000061
+			>;
+		};
+
+		pinctrl_lpspi0: lpspi0grp {
+			fsl,pins = <
+				IMX8DXL_SPI0_SCK_ADMA_SPI0_SCK					0x600004c
+				IMX8DXL_SPI0_SDO_ADMA_SPI0_SDO					0x600004c
+				IMX8DXL_SPI0_SDI_ADMA_SPI0_SDI					0x600004c
+				/* IMX8DXL_SPI0_CS0_ADMA_SPI0_CS0					0x600004c */
+				/* IMX8DXL_SPI0_CS1_ADMA_SPI0_CS1					0x600004c */
+				IMX8DXL_SPI0_CS0_LSIO_GPIO1_IO08				0x0000021
+				IMX8DXL_SPI0_CS1_LSIO_GPIO1_IO07				0x0000021
+			>;
+		};
+
+		pinctrl_lpspi0_switch: lpspi0switchgrp {
+			fsl,pins = <
+				/* SW_RSTn: io without pull-up */
+				IMX8DXL_USB_SS3_TC0_LSIO_GPIO4_IO03				0x0000021
+
+				/* SW_CORE_RSTn: io without pull-up */
+				IMX8DXL_USB_SS3_TC1_LSIO_GPIO4_IO04				0x0000021
+			>;
+		};
+
+		pinctrl_usdhc3: usdhc3grp {
+			fsl,pins = <
+				IMX8DXL_ENET0_RGMII_TXC_CONN_USDHC2_CLK			0x06000040
+				IMX8DXL_ENET0_RGMII_TX_CTL_CONN_USDHC2_CMD		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD0_CONN_USDHC2_DATA0		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD1_CONN_USDHC2_DATA1		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD2_CONN_USDHC2_DATA2		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD3_CONN_USDHC2_DATA3		0x00000021
+			>;
+		};
+
+		pinctrl_usdhc3_100mhz: usdhc3grp100mhz {
+			fsl,pins = <
+				IMX8DXL_ENET0_RGMII_TXC_CONN_USDHC2_CLK			0x06000040
+				IMX8DXL_ENET0_RGMII_TX_CTL_CONN_USDHC2_CMD		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD0_CONN_USDHC2_DATA0		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD1_CONN_USDHC2_DATA1		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD2_CONN_USDHC2_DATA2		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD3_CONN_USDHC2_DATA3		0x00000021
+			>;
+		};
+
+		pinctrl_usdhc3_200mhz: usdhc3grp200mhz {
+			fsl,pins = <
+				IMX8DXL_ENET0_RGMII_TXC_CONN_USDHC2_CLK			0x06000040
+				IMX8DXL_ENET0_RGMII_TX_CTL_CONN_USDHC2_CMD		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD0_CONN_USDHC2_DATA0		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD1_CONN_USDHC2_DATA1		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD2_CONN_USDHC2_DATA2		0x00000021
+				IMX8DXL_ENET0_RGMII_TXD3_CONN_USDHC2_DATA3		0x00000021
+			>;
+		};
+
+		pinctrl_wifi: wifigrp {
+			fsl,pins = <
+				/*
+				 * WL_REG_ON for Carrier revision 1.0:
+				 * SPI chipselect with pull-up, to force always high
+				 */
+				IMX8DXL_SPI3_CS1_ADMA_SPI3_CS1		0x0000021
+
+				/*
+				 * WL_REG_ON for Carrier revision 1.1 and later:
+				 * io without pull-up (WiFi module has internal pull-down)
+				 */
+				IMX8DXL_SPI3_SDI_LSIO_GPIO0_IO15		0x0000061
+			>;
+		};
+
+		pinctrl_bluetooth: bluetoothgrp {
+			fsl,pins = <
+				/* bt_reg_on: io without pull-up (BT module has internal pull-down) */
+				IMX8DXL_SPI3_SCK_LSIO_GPIO0_IO13	0x0000061
+			>;
+		};
+
+		pinctrl_reg_1v1: 1v1grp {
+			fsl,pins = <
+				/* SW_PE USB_OTG1_OC: io without pull-up */
+				IMX8DXL_USB_SS3_TC2_LSIO_GPIO4_IO05	0x0000061
+			>;
+		};
+};
+
+/* exposed on J14: SDA(51), SCL(53) */
+&i2c2 {
+	/* MFS5600AMMA7ES @ 18 */
+
+	/* TCA6408ARGTR */
+	tca6408: gpio@20 {
+		compatible = "ti,tca6408";
+		reg = <0x20>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		reset-gpios = <&lsio_gpio2 3 GPIO_ACTIVE_LOW>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_i2c2_gpio_rst>, <&pinctrl_i2c2_gpio_int>;
+		gpio-line-names = "DIG_IN1", "DIG_IN2", "CAN_STNB1", "CAN_STNB2", "DIG_OUT1", "DIG_OUT2", "", "";
+		/* max 100kHz bus speed */
+	};
+};
+
+/* exposed on J14: SDA(33), SCL(31) */
+&i2c3 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3>;
+	status = "okay";
+
+	/* nothing connected */
+};
+
+&lpspi0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpspi0>, <&pinctrl_lpspi0_switch>;
+	status = "okay";
+	cs-gpios = <&lsio_gpio1 8 GPIO_ACTIVE_LOW>, <&lsio_gpio1 7 GPIO_ACTIVE_LOW>;
+/*
+	spidev@0 {
+		compatible = "rohm,dh2228fv";
+		reg = <0>;
+		spi-max-frequency = <4000000>;
+	};
+
+	spidev@1 {
+		compatible = "rohm,dh2228fv";
+		reg = <1>;
+		spi-max-frequency = <4000000>;
+*/
+
+	ethernet-switch@0 {
+		compatible = "nxp,sja1110a";
+		reg = <0>;
+		//spi-max-frequency = <25000000>;
+		spi-max-frequency = <4000000>;
+		reset-gpios = <&lsio_gpio4 3 GPIO_ACTIVE_LOW>, <&lsio_gpio4 4 GPIO_ACTIVE_LOW>;
+
+		ethernet-ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			/* to Switch uC */
+			port@0 {
+				reg = <0x0>;
+				status = "disabled";
+				label = "uc";
+			};
+
+			/* to RJ45 connector on V2X Adapter, 100BASE-TX */
+			port@1 {
+				reg = <0x1>;
+				status = "okay";
+				label = "lan1";
+				phy-mode = "internal";
+				phy-handle = <&switch_port1_base_tx_phy>;
+			};
+
+			/* to CPU */
+			port@2 {
+				reg = <0x2>;
+				status = "okay";
+				label = "cpu";
+				phy-mode = "rgmii-id";
+				rx-internal-delay-ps = <2000>;
+				tx-internal-delay-ps = <2000>;
+				ethernet = <&eqos>;
+
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+				};
+			};
+
+			port@3 {
+				reg = <0x3>;
+				status = "disabled";
+				label = "lan3";
+			};
+
+			port@4 {
+				reg = <0x4>;
+				status = "disabled";
+				label = "lan4";
+			};
+
+			port@5 {
+				reg = <0x5>;
+				status = "disabled";
+				label = "trx1";
+				phy-mode = "internal";
+				phy-handle = <&switch_port5_base_t1_phy>;
+			};
+
+			port@6 {
+				reg = <0x6>;
+				status = "disabled";
+				label = "trx2";
+				phy-mode = "internal";
+				phy-handle = <&switch_port6_base_t1_phy>;
+			};
+
+			port@7 {
+				reg = <0x7>;
+				status = "disabled";
+				label = "trx3";
+				phy-mode = "internal";
+				phy-handle = <&switch_port7_base_t1_phy>;
+			};
+
+			port@8 {
+				reg = <0x8>;
+				status = "disabled";
+				label = "trx4";
+				phy-mode = "internal";
+				phy-handle = <&switch_port8_base_t1_phy>;
+			};
+
+			port@9 {
+				reg = <0x9>;
+				status = "disabled";
+				label = "trx5";
+				phy-mode = "internal";
+				phy-handle = <&switch_port9_base_t1_phy>;
+			};
+
+			port@a {
+				reg = <0xa>;
+				status = "okay";
+				label = "trx6";
+				phy-mode = "internal";
+				phy-handle = <&switch_port10_base_t1_phy>;
+			};
+		};
+
+		mdios {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			mdio@0 {
+				compatible = "nxp,sja1110-base-t1-mdio";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0>;
+				status = "okay";
+
+				switch_port5_base_t1_phy: ethernet-phy@1 {
+					compatible = "ethernet-phy-ieee802.3-c45";
+					reg = <0x1>;
+					status = "disabled";
+				};
+
+				switch_port6_base_t1_phy: ethernet-phy@2 {
+					compatible = "ethernet-phy-ieee802.3-c45";
+					reg = <0x2>;
+					status = "disabled";
+				};
+
+				switch_port7_base_t1_phy: ethernet-phy@3 {
+					compatible = "ethernet-phy-ieee802.3-c45";
+					reg = <0x3>;
+					status = "disabled";
+				};
+
+				switch_port8_base_t1_phy: ethernet-phy@4 {
+					compatible = "ethernet-phy-ieee802.3-c45";
+					reg = <0x4>;
+					status = "disabled";
+				};
+
+				switch_port9_base_t1_phy: ethernet-phy@5 {
+					compatible = "ethernet-phy-ieee802.3-c45";
+					reg = <0x5>;
+					status = "disabled";
+				};
+
+				switch_port10_base_t1_phy: ethernet-phy@6 {
+					compatible = "ethernet-phy-ieee802.3-c45";
+					reg = <0x6>;
+					status = "okay";
+				};
+			};
+
+			mdio@1 {
+				compatible = "nxp,sja1110-base-tx-mdio";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <1>;
+				status = "okay";
+
+				switch_port1_base_tx_phy: ethernet-phy@1 {
+					reg = <0x1>;
+					status = "okay";
+				};
+
+				switch_port2_base_tx_phy: ethernet-phy@2 {
+					reg = <0x2>;
+					status = "disabled";
+				};
+
+				switch_port3_base_tx_phy: ethernet-phy@3 {
+					reg = <0x3>;
+					status = "disabled";
+				};
+
+				switch_port4_base_tx_phy: ethernet-phy@4 {
+					reg = <0x4>;
+					status = "disabled";
+				};
+			};
+		};
+	};
+};
+
+/* bluetooth */
+&lpuart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpuart1>, <&pinctrl_bluetooth>;
+	status = "okay";
+	uart-has-rtscts;
+
+	bt: bluetooth {
+		compatible = "brcm,bcm4330-bt";
+		max-speed = <4000000>;
+		shutdown-gpios = <&lsio_gpio0 13 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&lsio_gpio0 {
+	status = "okay";
+};
+
+&lsio_gpio4 {
+	status = "okay";
+};
+
+&usbotg2 {
+	srp-disable;
+	hnp-disable;
+	adp-disable;
+	power-active-high;
+	disable-over-current;
+	status = "okay";
+	dr_mode = "host";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lte>;
+};
+
+&usbphy2 {
+	status = "okay";
+};
+
+/* WiFI (Carrier) */
+&usdhc3 {
+	// verify that REF_CLK_32K is enabled */
+    #address-cells = <1>;
+    #size-cells = <0>;
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc3>, <&pinctrl_wifi>;
+	pinctrl-1 = <&pinctrl_usdhc3_100mhz>, <&pinctrl_wifi>;
+	pinctrl-2 = <&pinctrl_usdhc3_200mhz>, <&pinctrl_wifi>;
+	bus-width = <4>;
+	no-sd;
+	non-removable;
+	status = "okay";
+	max-frequency = <40000000>;
+	fsl,delay-line = <12>;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	mmc-pwrseq = <&usdhc3_pwrseq>;
+
+	/* CYW43455 */
+    wifi: cyw43455@1 {
+		reg = <1>;
+		compatible = "brcm,bcm4329-fmac";
+    };
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8dxl-v2x.dts b/arch/arm64/boot/dts/freescale/imx8dxl-v2x.dts
index d3cb9dd497d3..afd62adc066c 100644
--- a/arch/arm64/boot/dts/freescale/imx8dxl-v2x.dts
+++ b/arch/arm64/boot/dts/freescale/imx8dxl-v2x.dts
@@ -1,565 +1,16 @@
 // SPDX-License-Identifier: GPL-2.0+
 /*
  * Copyright 2019~2020 NXP
- * Copyright 2022 Josua Mayer <josua@solid-run.com>
+ * Copyright 2022-2023 Josua Mayer <josua@solid-run.com>
  */
 
 /dts-v1/;
 
 #include "imx8dxl.dtsi"
 #include "imx8dxl-sr-som.dtsi"
+#include "imx8dxl-sr-v2x-gateway.dtsi"
 
 / {
-	model = "SolidRun i.MX8DXL V2X Carrier";
-	compatible = "fsl,imx8dxl-sr-som", "fsl,imx8dxl";
-
-	/*lte_vbat_reg: regulator-lte-vbat {
-		compatible = "regulator-fixed";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_lte_vbat>;
-		regulator-name = "lte-vbat";
-		regulator-min-microvolt = <3600000>;
-		regulator-max-microvolt = <3600000>;
-		gpio = <&lsio_gpio0 14 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-		regulator-always-on;
-	};*/
-
-	regulator_1v1: regulator-1v1 {
-		compatible = "regulator-fixed";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_reg_1v1>;
-		regulator-name = "1v1";
-		regulator-min-microvolt = <1100000>;
-		regulator-max-microvolt = <1100000>;
-		gpio = <&lsio_gpio4 5 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-		regulator-always-on;
-	};
-
-	regulator_flexcan1_standby: regulator-flexcan1-standby {
-		compatible = "regulator-fixed";
-		regulator-name = "flexcan1-standby";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		gpio = <&tca6408 2 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-	};
-
-	regulator_flexcan2_standby: regulator-flexcan2-standby {
-		compatible = "regulator-fixed";
-		regulator-name = "flexcan2-standby";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		gpio = <&tca6408 3 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-	};
-
-	usdhc3_pwrseq: usdhc3-pwrseq {
-		compatible = "mmc-pwrseq-simple";
-		reset-gpios = <&lsio_gpio0 15 GPIO_ACTIVE_LOW>;
-	};
-};
-
-&eqos {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_eqos>;
-	phy-connection-type = "rgmii-id";
-	status = "okay";
-
-	fixed-link {
-		speed = <1000>;
-		full-duplex;
-	};
-};
-
-&flexcan1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_flexcan1>;
-	xceiver-supply = <&regulator_flexcan1_standby>;
-	status = "okay";
-
-	can-transceiver {
-		/* TJA1048 */
-		max-bitrate = <5000000>;
-	};
-};
-
-&flexcan2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_flexcan2>;
-	xceiver-supply = <&regulator_flexcan2_standby>;
-	status = "okay";
-
-	can-transceiver {
-		/* TJA1048 */
-		max-bitrate = <5000000>;
-	};
-};
-
-&iomuxc {
-		pinctrl_eqos: eqosgrp {
-			fsl,pins = <
-				/* MDIO to Switch */
-				/* enet0 mdio pads supplied with 3.3v */
-				/* IMX8DXL_COMP_CTL_GPIO_1V8_3V3_GPIOCT */
-				IMX8DXL_ENET0_MDC_CONN_EQOS_MDC				0x06000020
-				IMX8DXL_ENET0_MDIO_CONN_EQOS_MDIO			0x06000020
-				/* RGMII to Switch */
-				/* ENET1 pad supplied with 3.3V */
-				/*IMX8DXL_COMP_CTL_GPIO_1V8_3V3_GPIORHB		0x000514a0*/
-				IMX8DXL_ENET1_RGMII_TX_CTL_CONN_EQOS_RGMII_TX_CTL	0x06000020
-				IMX8DXL_ENET1_RGMII_TXC_CONN_EQOS_RGMII_TXC		0x06000020
-				IMX8DXL_ENET1_RGMII_TXD0_CONN_EQOS_RGMII_TXD0		0x06000020
-				IMX8DXL_ENET1_RGMII_TXD1_CONN_EQOS_RGMII_TXD1		0x06000020
-				IMX8DXL_ENET1_RGMII_TXD2_CONN_EQOS_RGMII_TXD2		0x06000020
-				IMX8DXL_ENET1_RGMII_TXD3_CONN_EQOS_RGMII_TXD3		0x06000020
-				IMX8DXL_ENET1_RGMII_RXC_CONN_EQOS_RGMII_RXC		0x06000020
-				IMX8DXL_ENET1_RGMII_RX_CTL_CONN_EQOS_RGMII_RX_CTL	0x06000020
-				IMX8DXL_ENET1_RGMII_RXD0_CONN_EQOS_RGMII_RXD0		0x06000020
-				IMX8DXL_ENET1_RGMII_RXD1_CONN_EQOS_RGMII_RXD1		0x06000020
-				IMX8DXL_ENET1_RGMII_RXD2_CONN_EQOS_RGMII_RXD2		0x06000020
-				IMX8DXL_ENET1_RGMII_RXD3_CONN_EQOS_RGMII_RXD3		0x06000020
-			>;
-		};
-
-		pinctrl_flexcan1: flexcan1grp {
-			fsl,pins = <
-				IMX8DXL_FLEXCAN0_TX_ADMA_FLEXCAN0_TX	0x00000021
-				IMX8DXL_FLEXCAN0_RX_ADMA_FLEXCAN0_RX	0x00000021
-			>;
-		};
-
-		pinctrl_flexcan2: flexcan2grp {
-			fsl,pins = <
-				IMX8DXL_FLEXCAN1_TX_ADMA_FLEXCAN1_TX	0x00000021
-				IMX8DXL_FLEXCAN1_RX_ADMA_FLEXCAN1_RX	0x00000021
-			>;
-		};
-
-		pinctrl_i2c2_gpio: i2c2gpiogrp {
-			fsl,pins = <
-				/* gpio-expander reset: io without pull-up */
-				IMX8DXL_SCU_GPIO0_00_LSIO_GPIO2_IO03		0x0000061
-
-				/* gpio-expander interrupt: io without pull-up */
-				/* not usable, because this pin does not support lsio gpio mode :( */
-				IMX8DXL_SCU_GPIO0_01_SCU_GPIO0_IO01		0x0000061
-			>;
-		};
-
-		pinctrl_i2c3: i2c3grp {
-			fsl,pins = <
-				IMX8DXL_SPI1_CS0_ADMA_I2C3_SDA		0x06000021
-				IMX8DXL_SPI1_SDI_ADMA_I2C3_SCL		0x06000021
-			>;
-		};
-
-		pinctrl_lpuart1: lpuart1grp {
-			fsl,pins = <
-				IMX8DXL_UART1_RX_ADMA_UART1_RX			0x06000020
-				IMX8DXL_UART1_TX_ADMA_UART1_TX			0x06000020
-				IMX8DXL_UART1_CTS_B_ADMA_UART1_CTS_B	0x06000020
-				IMX8DXL_UART1_RTS_B_ADMA_UART1_RTS_B	0x06000020
-			>;
-		};
-
-		pinctrl_lte: ltegrp {
-			fsl,pins = <
-				/* modem RESET_N: io open drain drive 2mA */
-				IMX8DXL_ADC_IN3_LSIO_GPIO1_IO11	0x2000061
-
-				/* modem PWRKEY: io open drain drive 2mA */
-				IMX8DXL_ADC_IN2_LSIO_GPIO1_IO12	0x2000061
-			>;
-		};
-
-		pinctrl_lte_vbat: ltevbatgrp {
-			fsl,pins = <
-				/* RF_PWR: io without pull-up */
-				IMX8DXL_SPI3_SDO_LSIO_GPIO0_IO14	0x0000061
-			>;
-		};
-
-		pinctrl_lpspi0: lpspi0grp {
-			fsl,pins = <
-				IMX8DXL_SPI0_SCK_ADMA_SPI0_SCK					0x600004c
-				IMX8DXL_SPI0_SDO_ADMA_SPI0_SDO					0x600004c
-				IMX8DXL_SPI0_SDI_ADMA_SPI0_SDI					0x600004c
-				/* IMX8DXL_SPI0_CS0_ADMA_SPI0_CS0					0x600004c */
-				/* IMX8DXL_SPI0_CS1_ADMA_SPI0_CS1					0x600004c */
-				IMX8DXL_SPI0_CS0_LSIO_GPIO1_IO08				0x0000021
-				IMX8DXL_SPI0_CS1_LSIO_GPIO1_IO07				0x0000021
-			>;
-		};
-
-		pinctrl_lpspi0_switch: lpspi0switchgrp {
-			fsl,pins = <
-				/* SW_RSTn: io without pull-up */
-				IMX8DXL_USB_SS3_TC0_LSIO_GPIO4_IO03				0x0000021
-
-				/* SW_CORE_RSTn: io without pull-up */
-				IMX8DXL_USB_SS3_TC1_LSIO_GPIO4_IO04				0x0000021
-			>;
-		};
-
-		pinctrl_usdhc3: usdhc3grp {
-			fsl,pins = <
-				IMX8DXL_ENET0_RGMII_TXC_CONN_USDHC2_CLK			0x06000040
-				IMX8DXL_ENET0_RGMII_TX_CTL_CONN_USDHC2_CMD		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD0_CONN_USDHC2_DATA0		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD1_CONN_USDHC2_DATA1		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD2_CONN_USDHC2_DATA2		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD3_CONN_USDHC2_DATA3		0x00000021
-			>;
-		};
-
-		pinctrl_usdhc3_100mhz: usdhc3grp100mhz {
-			fsl,pins = <
-				IMX8DXL_ENET0_RGMII_TXC_CONN_USDHC2_CLK			0x06000040
-				IMX8DXL_ENET0_RGMII_TX_CTL_CONN_USDHC2_CMD		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD0_CONN_USDHC2_DATA0		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD1_CONN_USDHC2_DATA1		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD2_CONN_USDHC2_DATA2		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD3_CONN_USDHC2_DATA3		0x00000021
-			>;
-		};
-
-		pinctrl_usdhc3_200mhz: usdhc3grp200mhz {
-			fsl,pins = <
-				IMX8DXL_ENET0_RGMII_TXC_CONN_USDHC2_CLK			0x06000040
-				IMX8DXL_ENET0_RGMII_TX_CTL_CONN_USDHC2_CMD		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD0_CONN_USDHC2_DATA0		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD1_CONN_USDHC2_DATA1		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD2_CONN_USDHC2_DATA2		0x00000021
-				IMX8DXL_ENET0_RGMII_TXD3_CONN_USDHC2_DATA3		0x00000021
-			>;
-		};
-
-		pinctrl_wifi: wifigrp {
-			fsl,pins = <
-				/*
-				 * WL_REG_ON for Carrier revision 1.0:
-				 * SPI chipselect with pull-up, to force always high
-				 */
-				IMX8DXL_SPI3_CS1_ADMA_SPI3_CS1		0x0000021
-
-				/*
-				 * WL_REG_ON for Carrier revision 1.1 and later:
-				 * io without pull-up (WiFi module has internal pull-down)
-				 */
-				IMX8DXL_SPI3_SDI_LSIO_GPIO0_IO15		0x0000061
-			>;
-		};
-
-		pinctrl_bluetooth: bluetoothgrp {
-			fsl,pins = <
-				/* bt_reg_on: io without pull-up (BT module has internal pull-down) */
-				IMX8DXL_SPI3_SCK_LSIO_GPIO0_IO13	0x0000061
-			>;
-		};
-
-		pinctrl_reg_1v1: 1v1grp {
-			fsl,pins = <
-				/* SW_PE USB_OTG1_OC: io without pull-up */
-				IMX8DXL_USB_SS3_TC2_LSIO_GPIO4_IO05	0x0000061
-			>;
-		};
-};
-
-/* exposed on J14: SDA(51), SCL(53) */
-&i2c2 {
-	/* MFS5600AMMA7ES @ 18 */
-
-	/* TCA6408ARGTR */
-	tca6408: gpio@20 {
-		compatible = "ti,tca6408";
-		reg = <0x20>;
-		gpio-controller;
-		#gpio-cells = <2>;
-		reset-gpios = <&lsio_gpio2 3 GPIO_ACTIVE_LOW>;
-		//interrupt-parent = <&lsio_gpio0>;
-        //interrupts = <1 IRQ_TYPE_EDGE_FALLING>; /* re-enable after changing layout */
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_i2c2_gpio>;
-		gpio-line-names = "DIG_IN1", "DIG_IN2", "CAN_STNB1", "CAN_STNB2", "DIG_OUT1", "DIG_OUT2", "", "";
-		/* max 100kHz bus speed */
-	};
-};
-
-/* exposed on J14: SDA(33), SCL(31) */
-&i2c3 {
-	#address-cells = <1>;
-	#size-cells = <0>;
-	clock-frequency = <100000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c3>;
-	status = "okay";
-
-	/* nothing connected */
-};
-
-&lpspi0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lpspi0>, <&pinctrl_lpspi0_switch>;
-	status = "okay";
-	cs-gpios = <&lsio_gpio1 8 GPIO_ACTIVE_LOW>, <&lsio_gpio1 7 GPIO_ACTIVE_LOW>;
-/*
-	spidev@0 {
-		compatible = "rohm,dh2228fv";
-		reg = <0>;
-		spi-max-frequency = <4000000>;
-	};
-
-	spidev@1 {
-		compatible = "rohm,dh2228fv";
-		reg = <1>;
-		spi-max-frequency = <4000000>;
-*/
-
-	ethernet-switch@0 {
-		compatible = "nxp,sja1110a";
-		reg = <0>;
-		//spi-max-frequency = <25000000>;
-		spi-max-frequency = <4000000>;
-		reset-gpios = <&lsio_gpio4 3 GPIO_ACTIVE_LOW>, <&lsio_gpio4 4 GPIO_ACTIVE_LOW>;
-
-		ethernet-ports {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			/* to Switch uC */
-			port@0 {
-				reg = <0x0>;
-				status = "disabled";
-				label = "uc";
-			};
-
-			/* to RJ45 connector on V2X Adapter, 100BASE-TX */
-			port@1 {
-				reg = <0x1>;
-				status = "okay";
-				label = "lan1";
-				phy-mode = "internal";
-				phy-handle = <&switch_port1_base_tx_phy>;
-			};
-
-			/* to CPU */
-			port@2 {
-				reg = <0x2>;
-				status = "okay";
-				label = "cpu";
-				phy-mode = "rgmii-id";
-				rx-internal-delay-ps = <2000>;
-				tx-internal-delay-ps = <2000>;
-				ethernet = <&eqos>;
-
-				fixed-link {
-					speed = <1000>;
-					full-duplex;
-				};
-			};
-
-			port@3 {
-				reg = <0x3>;
-				status = "disabled";
-				label = "lan3";
-			};
-
-			port@4 {
-				reg = <0x4>;
-				status = "disabled";
-				label = "lan4";
-			};
-
-			port@5 {
-				reg = <0x5>;
-				status = "disabled";
-				label = "trx1";
-				phy-mode = "internal";
-				phy-handle = <&switch_port5_base_t1_phy>;
-			};
-
-			port@6 {
-				reg = <0x6>;
-				status = "disabled";
-				label = "trx2";
-				phy-mode = "internal";
-				phy-handle = <&switch_port6_base_t1_phy>;
-			};
-
-			port@7 {
-				reg = <0x7>;
-				status = "disabled";
-				label = "trx3";
-				phy-mode = "internal";
-				phy-handle = <&switch_port7_base_t1_phy>;
-			};
-
-			port@8 {
-				reg = <0x8>;
-				status = "disabled";
-				label = "trx4";
-				phy-mode = "internal";
-				phy-handle = <&switch_port8_base_t1_phy>;
-			};
-
-			port@9 {
-				reg = <0x9>;
-				status = "disabled";
-				label = "trx5";
-				phy-mode = "internal";
-				phy-handle = <&switch_port9_base_t1_phy>;
-			};
-
-			port@a {
-				reg = <0xa>;
-				status = "okay";
-				label = "trx6";
-				phy-mode = "internal";
-				phy-handle = <&switch_port10_base_t1_phy>;
-			};
-		};
-
-		mdios {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			mdio@0 {
-				compatible = "nxp,sja1110-base-t1-mdio";
-				#address-cells = <1>;
-				#size-cells = <0>;
-				reg = <0>;
-				status = "okay";
-
-				switch_port5_base_t1_phy: ethernet-phy@1 {
-					compatible = "ethernet-phy-ieee802.3-c45";
-					reg = <0x1>;
-					status = "disabled";
-				};
-
-				switch_port6_base_t1_phy: ethernet-phy@2 {
-					compatible = "ethernet-phy-ieee802.3-c45";
-					reg = <0x2>;
-					status = "disabled";
-				};
-
-				switch_port7_base_t1_phy: ethernet-phy@3 {
-					compatible = "ethernet-phy-ieee802.3-c45";
-					reg = <0x3>;
-					status = "disabled";
-				};
-
-				switch_port8_base_t1_phy: ethernet-phy@4 {
-					compatible = "ethernet-phy-ieee802.3-c45";
-					reg = <0x4>;
-					status = "disabled";
-				};
-
-				switch_port9_base_t1_phy: ethernet-phy@5 {
-					compatible = "ethernet-phy-ieee802.3-c45";
-					reg = <0x5>;
-					status = "disabled";
-				};
-
-				switch_port10_base_t1_phy: ethernet-phy@6 {
-					compatible = "ethernet-phy-ieee802.3-c45";
-					reg = <0x6>;
-					status = "okay";
-				};
-			};
-
-			mdio@1 {
-				compatible = "nxp,sja1110-base-tx-mdio";
-				#address-cells = <1>;
-				#size-cells = <0>;
-				reg = <1>;
-				status = "okay";
-
-				switch_port1_base_tx_phy: ethernet-phy@1 {
-					reg = <0x1>;
-					status = "okay";
-				};
-
-				switch_port2_base_tx_phy: ethernet-phy@2 {
-					reg = <0x2>;
-					status = "disabled";
-				};
-
-				switch_port3_base_tx_phy: ethernet-phy@3 {
-					reg = <0x3>;
-					status = "disabled";
-				};
-
-				switch_port4_base_tx_phy: ethernet-phy@4 {
-					reg = <0x4>;
-					status = "disabled";
-				};
-			};
-		};
-	};
-};
-
-/* bluetooth */
-&lpuart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lpuart1>, <&pinctrl_bluetooth>;
-	status = "okay";
-	uart-has-rtscts;
-
-	bt: bluetooth {
-		compatible = "brcm,bcm4330-bt";
-		max-speed = <4000000>;
-		shutdown-gpios = <&lsio_gpio0 13 GPIO_ACTIVE_HIGH>;
-	};
-};
-
-&lsio_gpio0 {
-	status = "okay";
-};
-
-&lsio_gpio4 {
-	status = "okay";
-};
-
-&usbotg2 {
-	srp-disable;
-	hnp-disable;
-	adp-disable;
-	power-active-high;
-	disable-over-current;
-	status = "okay";
-	dr_mode = "host";
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lte>;
-};
-
-&usbphy2 {
-	status = "okay";
-};
-
-/* WiFI (Carrier) */
-&usdhc3 {
-	// verify that REF_CLK_32K is enabled */
-    #address-cells = <1>;
-    #size-cells = <0>;
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-	pinctrl-0 = <&pinctrl_usdhc3>, <&pinctrl_wifi>;
-	pinctrl-1 = <&pinctrl_usdhc3_100mhz>, <&pinctrl_wifi>;
-	pinctrl-2 = <&pinctrl_usdhc3_200mhz>, <&pinctrl_wifi>;
-	bus-width = <4>;
-	no-sd;
-	non-removable;
-	status = "okay";
-	max-frequency = <40000000>;
-	fsl,delay-line = <12>;
-	keep-power-in-suspend;
-	enable-sdio-wakeup;
-	mmc-pwrseq = <&usdhc3_pwrseq>;
-
-	/* CYW43455 */
-    wifi: cyw43455@1 {
-		reg = <1>;
-		compatible = "brcm,bcm4329-fmac";
-    };
+	model = "SolidRun i.MX8DXL V2X Gateway";
+	compatible = "solidrun,v2x-gateway", "fsl,imx8dxl";
 };
-- 
2.43.0

